Wbooks/BookRepository.swift

```diff
@@ -55,22 +58,70 @@ struct BookRepository: BookRepositoryType, RentRepositoryType {
                 onError(error)
                 return
             }
-            
-            
-            
+        }
+    }
+    
+    func fetchComments(bookID: Int, onSuccess: @escaping ([Comment]) -> Void, onError: @escaping (Error) -> Void) {
+        let url = URL(string: BookRepository.baseUrl + BookRepository.books + "/\(bookID)" + BookRepository.comments)!
+        AF.request(url, method: .get).responseJSON { response in
+            switch response.result {
+            case .success(let value):
+                guard let JSONcomments = try? JSONSerialization.data(withJSONObject: value, options: []),
+                      let comments = try? JSONDecoder().decode([Comment].self, from: JSONcomments) else {
+                    onError(CommentError.decodeError)
+                    return
+                }
+                onSuccess(comments)
+            case .failure(let error):
+                onError(error)
+            }
+        }
+    }
+    
+    func fetchUser(userID: Int, onSuccess: @escaping (User) -> Void, onError: @escaping (Error) -> Void) {
+        let url = URL(string: BookRepository.baseUrl + BookRepository.users + "/\(userID)")!
+        AF.request(url, method: .get).responseJSON { response in
+            switch response.result {
+            case.success(let value):
+                guard let JSONuser = try? JSONSerialization.data(withJSONObject: value, options: []), let user = try? JSONDecoder().decode(User.self, from: JSONuser) else {
+                    onError(UserError.decodeError)
+                    return
+                }
+                onSuccess(user)
+            case .failure(let error):
+                onError(error)
+            }
         }
     }
 
+    // MARK: Errors
     enum BookError: Error {
         case decodeError
     }
+    
+    enum CommentError: Error {
+        case decodeError
+    }
+    
+    enum UserError: Error {
+        case decodeError
+    }
```

Comment:
If you prefer, you can use one type of error instead of one for each entity. For example:
```Swift
enum RepositoryError: Error {
   case decodeError
}
```